internal/controlplane/handlers_artifacts.go

```diff
@@ -42,6 +42,10 @@ func (s *Server) ListArtifacts(ctx context.Context, in *pb.ListArtifactsRequest)
 	projectID := entityCtx.Project.ID
 	providerName := entityCtx.Provider.Name
 
+	// Telemetry logging
```

Comment:
In a few places, I moved the logging earlier in the method so that we capture the data sooner